### PR TITLE
Mk kolejka communication

### PR DESCRIPTION
**Pull request by m-kadula**

### CODE REVIEW

File: broker/tests.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+from broker.master import KolejkaCommunicationServer
+from http import client
+from threading import Thread, Lock
+import random as rand
+from time import sleep
+
+
+def test1_kolejka_communication(sub_number=1000, host='127.0.0.1', port=8080, max_wait=0.01) -> int:
```
Comment (ZyndramZM) [inline broker/tests.py]: Trzeba przerobić to na klasę unittestów. Link do dokumentacji: [Unittesty w Pythonie](https://docs.python.org/3/library/unittest.html). Przykład tego jak to robić jest w oryginalnym repo BaCy2.

File: broker/tests.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+from broker.master import KolejkaCommunicationServer
+from http import client
+from threading import Thread, Lock
+import random as rand
```
Comment (ZyndramZM) [inline broker/tests.py]: Testy nie powinny być randomizowane - powtarzalność pozwala znaleźć błędy

File: broker/tests.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    check = Thread(target=check_submits)
+    wait = Thread(target=wait_for_submits)
+    send.start()
+    check.start()
+    wait.start()
+
+    send.join()
+    check.join()
+    wait.join()
+    server.stop_server()
```
Comment (ZyndramZM) [inline broker/tests.py]: Najlepiej napisać testy sprawdzające po kolei każdy segment i 1-2 testy zbiorowe testujące całość. Pamiętaj o rozważeniu edge-caseów. Co się stanie jak broker dostanie niepoprawne dane na wejściu? To też powinno być sprawdzone w testach

File: broker/master.py [hunk 4]
```diff
@@ -29,3 +31,71 @@ def close_submit(self, submit_id: str):
...
+        with self.integrity_lock:
+            self.submit_dict[submit_id] = Lock()
+            self.submit_dict[submit_id].acquire()
+
+    def release_submit(self, submit_id: str) -> None:
+        with self.integrity_lock:
+            if self.submit_dict[submit_id].locked():
+                self.submit_dict[submit_id].release()
+            else:
+                raise ValueError  # TODO
```
Comment (ZyndramZM) [inline broker/master.py]: Powinien być na to jakiś błąd w bibliotece Threading - warto rzucić właśnie nim

**Review by ZyndramZM — CHANGES_REQUESTED** [review]
Plik tests.py wyrzućmy do osobnego folderu (tests) i zmieńmy jego nazwę, tak by sugerowała jaka część kodu jest testowana

**Review by ZyndramZM — APPROVED** [review]
